flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -35,6 +40,112 @@ def test_dataset_creation(self, cluster):
         """
         return create_dataset(self, cluster)
 
+    def _get_package_source(self, default_version=None):
+        """
+        Get the package source for the flocker version under test from
+        environment variables.
+
+        :param unicode default_version: The version of flocker to use
+            if not specified in environment variables.
+
+        :return: A ``PackageSource`` that can be used to install the version of
+            flocker under test.
+        """
+        env_vars = ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_VERSION',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER']
+        missing_vars = list(x for x in env_vars if x not in os.environ)
+        if missing_vars:
+            raise SkipTest(
+                'Missing environment variables for upgrade test: %s.' %
+                ', '.join(missing_vars))
+        version = (os.environ['FLOCKER_ACCEPTANCE_PACKAGE_VERSION'] or
+                   default_version)
+        return PackageSource(
+            version=version,
+            branch=os.environ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH'],
+            build_server=os.environ['FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER'])
+
+    @skip_backend(
+        unsupported={DatasetBackend.loopback},
+        reason="Does not maintain compute_instance_id across restarting "
+               "flocker (and didn't as of most recent release).")
+    @run_test_with(async_runner(timeout=timedelta(minutes=6)))
+    @require_cluster(1)
+    def test_upgrade(self, cluster):
+        """
+        Given a dataset created and used with the previously installable
+        version of flocker, uninstalling the previous version of flocker and
+        installing HEAD does not destroy the data on the dataset.
+        """
+        node = cluster.nodes[0]
+        SAMPLE_STR = '123456' * 100
+
+        upgrade_from_version = get_installable_version(version)
```

Comment:
:question: `version` confused me for a while. This is the module imported `version` rather than the local `version` used in nearby functions. Might have been clearer to ``from flocker import __version__ as INSTALLED_FLOCKER_VERSION`` .